src/rocks_init.cpp

```diff
@@ -116,9 +116,7 @@ namespace mongo {
                 return builder.obj();
             }
 
-            bool supportsReadOnly() const final {
-                return true;
-            }
+            bool supportsReadOnly() const final { return true; }
```

Comment:
Return false for now, since you assert that Rocks was not opened in the read-only mode. Potentially also create a task to support read-only mode.

Reply:
done, I will record this in the issue board if possible to merge.